evalml/model_understanding/graphs.py

```diff
@@ -501,6 +501,46 @@ def graph_binary_objective_vs_threshold(pipeline, X, y, objective, steps=100):
     return _go.Figure(layout=layout, data=data)
 
 
+def _get_feature_names_from_str_or_col_index(X, names_or_col_indices):
+    """Helper function to map the user-input features param to column names."""
+    feature_list = []
+    for name_or_index in names_or_col_indices:
+        if isinstance(name_or_index, int):
+            feature_list.append(X.columns[name_or_index])
+        else:
+            feature_list.append(name_or_index)
+    return feature_list
+
+
+def _raise_value_error_if_any_features_all_nan(df):
+    """Helper for partial dependence data validation."""
+
+    nan_pct = df.isna().mean()
+    all_nan = nan_pct[nan_pct == 1].index.tolist()
+    all_nan = [f"'{name}'" for name in all_nan]
+
+    if all_nan:
+        raise ValueError("The following features have all NaN values and so the "
+                         f"partial dependence cannot be computed: {', '.join(all_nan)}")
+
+
+def _raise_value_error_if_mostly_one_value(df, percentile):
+    """Helper for partial dependence data validation."""
+    one_value = []
+    values = []
+
+    for col in df.columns:
+        normalized_counts = df[col].value_counts(normalize=True) + 0.01
+        normalized_counts = normalized_counts[normalized_counts > percentile]
+        if not normalized_counts.empty:
+            one_value.append(f"'{col}'")
+            values.append(str(normalized_counts.index[0]))
+
+    if one_value:
+        raise ValueError(f"Features {', '.join(one_value)} are mostly one value, ({', '.join(values)}), "
```

Comment:
nit-pick, but can we put parentheses around the feature names too?
```python
f"Features ({', '.join(one_value)}) are mostly
```
I think it's cleaner when we have multiple features

Reply:
I think this is also another good time to add the raises column to the docstring of the partial dependence function so people can see the linkage of this ValueError to the upper percentile.  It took me a second, so I'm sure it'll cause a bit of head scratching.

Reply:
Good point @chukarsten . I'll add that now!